src/infrastructure/persistence/PsqlUserRepository.test.ts

```diff
@@ -17,28 +26,44 @@ describe('PsqlUserRepository', () => {
   });
 
   it('inserts new and retrieves a user by id', async () => {
-    const user = await psqlUserRepository.save(new User(new UserId(), new Email('kostas1@example.com')));
+    const user = await psqlUserRepository.save(User.create(magicLinkAuthenticationDetails('kostas1@example.com')));
 
     const persistedUser = await psqlUserRepository.findByUserId(user.id);
 
     expect(persistedUser?.id).toEqual(user.id);
     expect(persistedUser?.email).toEqual(user.email);
+    expect(persistedUser?.authenticationDetails.value.value).toBe('kostas1@example.com');
```

Comment:
.value.value ?

Reply:
What would you suggest

Reply:
Yup, it's the `value` of `AuthenticationDetails`' `value` field.

Reply:
What if the 'value' was called 'identifier', so that it was `AuthenticationDetails.identifier.value`?